### DISCUSSION TITLE
Allow configuring canister_args_file in dfx.json

### DISCUSSION THREAD

**Discussion by viviveevee** [dfinity/sdk#3060] [Feature Requests]
Request by @bitdivine [on the forum](https://forum.dfinity.org/t/dfx-json-actor-with-mandatory-arguments/14454/10):

> It would actually be really useful to be able to specify a file containing arguments. Why a file? Because then the arguments can be created by the build command, or modified by the user, without having to change dfx.json. Is there any chance that we can get something like a canister_args_file in dfx.json?

**Comment by alejandrade** [comment]
I would like to second this request

**Comment by viviveevee** [comment]
Importing comment from duplicate #3619:

> It is convenient to store big init args in `init_args.did` file. Would be useful to be able to specify this file in dfx.json like this:
`"init_arg": "./init_arg.did",`
or
`"init_arg_file": "./init_arg.did",`
